engine/src/execution_graph/logic_controllers/BatchProcessing.cpp

```diff
@@ -278,6 +289,11 @@ kstatus BindableTableScan::run() {
             //this will allow us to prevent from having too many open file handles by being
             //able to limit the number of file tasks
             auto handle = provider->get_next(true);
+            if (provider->is_sql()) {
```

Comment:
you should not be adding any special logic here. This part of the code needs to be completely ignorant of what kind of provider it is

Reply:
done